File: app/src/main/java/com/example/chris/atp_music_player/ui/FavoritesFragment.java
package com.example.chris.atp_music_player.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.chris.atp_music_player.R;
import com.example.chris.atp_music_player.adapters.FavoritesListAdapter;
import com.example.chris.atp_music_player.models.Song;

import java.util.ArrayList;
import java.util.List;

import io.realm.Realm;
import io.realm.RealmResults;

public class FavoritesFragment extends ATPDataFragment {

    public static FavoritesFragment newInstance() {
        return new FavoritesFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        getActivity().setTitle(R.string.menu_favs);

        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    protected void loadData() {
        Realm realm = Realm.getInstance(getActivity());
        RealmResults<Song> favorites = realm.allObjects(Song.class);

        List<Song> songList = new ArrayList<>();

        for (Song song : favorites) {
            songList.add(song);
        }

        if (songList.isEmpty()) {
            mErrorView.setText(R.string.err_favs);
            mRecyclerView.setVisibility(View.GONE);
            mErrorView.setVisibility(View.VISIBLE);
        } else {
            mRecyclerView.setAdapter(new FavoritesListAdapter(getActivity(), songList));
        }
    }
}

File: app/src/main/java/com/example/chris/atp_music_player/ui/FavoritesFragment.kt
package com.example.chris.atp_music_player.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.chris.atp_music_player.R
import com.example.chris.atp_music_player.Song
import com.example.chris.atp_music_player.SongRequestModel
import com.example.chris.atp_music_player.adapters.FavoritesListAdapter
import com.example.chris.atp_music_player.playback.SongRequest
import io.realm.Realm
import kotlinx.android.synthetic.main.no_data.*
import kotlinx.android.synthetic.main.song_list.*

class FavoritesFragment : ATPListFragment() {
    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        activity.setTitle(R.string.menu_favs)
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun loadData() {
        val favorites = Realm.getInstance(context).allObjects(Song::class.java)
        val songList = favorites.map { song -> song }

        if (songList.isEmpty()) {
            error.setText(R.string.err_favs)
            recycler_view.visibility = View.GONE
            error.visibility = View.VISIBLE
        } else {
            recycler_view.adapter = FavoritesListAdapter(songList, { songs, position ->
                SongRequest.post(SongRequestModel(songs, position, true))
            })
        }
    }
}
